hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java

```diff
@@ -150,9 +151,14 @@ public synchronized void reportSinkSuccess(SinkPeer sinkPeer) {
    */
   public synchronized void chooseSinks() {
     List<ServerName> slaveAddresses = endpoint.getRegionServers();
```

Comment:
Further down in `setRegionServers(fetchSlavesAddresses(this.getZkw()))`, `fetchSlavesAddresses()`is not returning null, but empty list. So maybe we can safely log warning for `if(slaveAddresses.size()==0)` ?